test/storage/index_builder_test.cpp

```diff
@@ -0,0 +1,179 @@
+#include <string>
+#include <vector>
+#include "catalog/catalog.h"
+#include "catalog/catalog_sql_table.h"
+#include "storage/index/index_factory.h"
+#include "storage/index/index_manager.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier::storage::index {
+struct IndexBuilderTest : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+    index_manager_ = new IndexManager();
+    txn_ = txn_manager_->BeginTransaction();
+    catalog_ = new catalog::Catalog(txn_manager_, txn_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    TerrierTest::TearDown();
+    delete catalog_;  // delete catalog first
+    delete index_manager_;
+    delete txn_manager_;
+    delete txn_;
+  }
+
+  catalog::Catalog *catalog_;
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+
+  transaction::TransactionContext *txn_;
+  transaction::TransactionManager *txn_manager_;
+  IndexManager *index_manager_;
+};
+
+// Check the basic correctness of index populator
+// NOLINTNEXTLINE
+TEST_F(IndexBuilderTest, BasicCorrectnessTest) {
+  // terrier has db_oid_t DEFAULT_DATABASE_OID
+  const catalog::db_oid_t terrier_oid(catalog::DEFAULT_DATABASE_OID);
+  auto db_handle = catalog_->GetDatabaseHandle();
+  auto table_handle = db_handle.GetNamespaceHandle(txn_, terrier_oid).GetTableHandle(txn_, "public");
+
+  // define schema
+  std::vector<catalog::Schema::Column> cols;
+  cols.emplace_back("sex", type::TypeId::BOOLEAN, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("id", type::TypeId::INTEGER, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("name", type::TypeId::VARCHAR, 100, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("address", type::TypeId::VARCHAR, 100, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  catalog::Schema schema(cols);
+
+  // create table
+  auto table = table_handle.CreateTable(txn_, schema, "test_table");
+  auto table_entry = table_handle.GetTableEntry(txn_, "test_table");
+  EXPECT_NE(table_entry, nullptr);
+  std::string_view str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(0));
+  EXPECT_EQ(str, "public");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(1));
+  EXPECT_EQ(str, "test_table");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(2));
+  EXPECT_EQ(str, "pg_default");
+
+  // Insert a few rows into the table
+  auto ptr = table_handle.GetTable(txn_, "test_table");
+  EXPECT_EQ(ptr, table);
+
+  for (int i = 0; i < 100; ++i) {
+    std::vector<type::TransientValue> row;
+    std::ostringstream stringStream;
+    stringStream << "name_" << i;
+    std::string name_string = stringStream.str();
+    stringStream << "address_" << i;
```

Comment:
As said in the previous code review, I suggest using `fmt::format` instead of `stringstream`. since it is more straightforward.
And there are some other unaddressed code review comments in this file too. Please have a look at them.